### (a) Estimate the probability that a student who studies for 40 h and has an undergrad GPA of 3.5 gets an A in the class.


In [1]:
import numpy as np

# Given coefficients
beta_0 = -6
beta_1 = 0.05
beta_2 = 1

# Given data for part (a)
hours_studied = 40
gpa = 3.5

# Logistic regression formula: P(Y=1) = 1 / (1 + exp(-(β0 + β1 * X1 + β2 * X2)))
linear_combination = beta_0 + beta_1 * hours_studied + beta_2 * gpa
probability_a = 1 / (1 + np.exp(-linear_combination))

print(f"Probability of getting an A: {probability_a:.4f}")


Probability of getting an A: 0.3775


For a student who studied for 40 hours and has an undergraduate GPA of 3.5, the predicted probability of receiving an A, according to the logistic regression model, is 37.75%.

### (b) How many hours would the student in part (a) need to study to have a 50 % chance of getting an A in the class?


In [2]:
import numpy as np

# Given coefficients
beta_0 = -6
beta_1 = 0.05
beta_2 = 1

# Given GPA from part (a)
gpa = 3.5

# Target probability for 50% chance
target_probability = 0.5
log_odds_target = np.log(target_probability / (1 - target_probability))  # log(0.5 / 0.5) = 0

# Rearrange the logistic regression formula to solve for hours_studied
# 0 = β0 + β1 * hours_studied + β2 * gpa
# hours_studied = (log_odds_target - β0 - β2 * gpa) / β1

required_hours = (log_odds_target - beta_0 - beta_2 * gpa) / beta_1
print(f"Required study hours for 50% chance of getting an A: {required_hours:.2f} hours")


Required study hours for 50% chance of getting an A: 50.00 hours



Based on the logistic regression model, the student would need to study 50 hours to have a 50% chance of getting an A in the class.